<!DOCTYPE html>
<html lang="en-US">
  <head>
    <meta charset="utf-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <meta name="author" content="Yue Liu" />
    
    <link rel="shortcut icon" type="image/x-icon" href="/img/favicon.ico">
    <title>Project 1: Exploratory Data Analysis</title>
    <meta name="generator" content="Hugo 0.83.1" />
    <link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
    <link rel="stylesheet" type="text/css" href="/css/main.css" />
    <link rel="stylesheet" type="text/css" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.7.0/css/font-awesome.min.css" />
    <link rel="stylesheet" type="text/css" href="https://fonts.googleapis.com/css?family=Source+Sans+Pro:200,400,200bold,400old" />
    
    <!--[if lt IE 9]>
			<script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
			<script src="https://oss.maxcdn.com/libs/respond.js/1.4.2/respond.min.js"></script>
		<![endif]-->

    
  </head>

  <body>
    <div id="wrap">
      
      <nav class="navbar navbar-default">
  <div class="container">
    <div class="navbar-header">
      <a class="navbar-brand" href="/"><i class="fa fa-home"></i></a>
      <button type="button" class="navbar-toggle" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
    </div>
    <div class="navbar-collapse collapse" id="navbar">
      <ul class="nav navbar-nav navbar-right">
      
        
        <li><a href="/post/">BLOG</a></li>
        
        <li><a href="/projects/">PROJECTS</a></li>
        
        <li><a href="/resume/">RESUME</a></li>
        
      
      </ul>
    </div>
  </div>
</nav>

      <div class="container">
        <div class="blog-post">
          <h3>
            <strong><a href="/project/project1/">Project 1: Exploratory Data Analysis</a></strong>
          </h3>
        </div>
 
<div class="blog-title">
          <h4>
         January 1, 0001 
            &nbsp;&nbsp;
            
          </h4>
        </div>

        <div class="panel panel-default">
          <div class="panel-body">
            <div class="blogpost">
              


<div id="data-wrangling-and-data-exploration" class="section level2">
<h2>Data Wrangling and Data Exploration</h2>
<div id="instructions" class="section level3">
<h3>Instructions</h3>
<p>A knitted R Markdown document (ideally HTML) and the raw R Markdown file (as .Rmd) should both be submitted to Canvas by 11:59pm on the due date. These two documents will be graded jointly, so they must be consistent (i.e., don’t change the R Markdown file without also updating the knitted document).</p>
<p>The text of the document should provide a narrative structure around your code/output. All results presented must have corresponding code. Any answers/results/plots etc. given without the corresponding R code that generated the result will not be considered. Furthermore, all code contained in your final project document must work correctly (knit early, knit often)! Please do not include any extraneous code or code which produces error messages. (Code that produces warnings is acceptable, as long as you understand what the warnings mean!)</p>
</div>
<div id="find-data" class="section level3">
<h3>Find data:</h3>
<p>Find two (!) datasets with one variable in common (e.g., dates, times, states, counties, countries, sports players), both with at least 50 observations (i.e., rows) in each. Please think very carefully about whether it makes sense to combine your datasets! If you find one dataset with 50 patients and it has their age, and you find another dataset with 50 <em>different</em> patients that has their ages, it makes no sense to join them based on age (you would just be pairing up random people of the same age).</p>
<p>When combined, the resulting/final dataset must have <strong>at least 4 different variables (at least 3 numeric) in addition to the common variable</strong> (i.e., five variables total).</p>
<p>You can have as many variables as you would like! If you found two datasets that you like but they don’t have enough variables, find a third dataset with the same common variable and join all three.</p>
</div>
<div id="guidelines" class="section level3">
<h3>Guidelines</h3>
<ol style="list-style-type: decimal">
<li><p>If the datasets are not tidy, you will need to reshape them so that every observation has its own row and every variable its own column. If the datasets are both already tidy, you will make them untidy with <code>pivot_wider()/spread()</code> and then tidy them again with <code>pivot_longer/gather()</code> to demonstrate your use of the functions. It’s fine to wait until you have your descriptives to use these functions (e.g., you might want to pivot_wider() to rearrange the data to make your descriptive statistics easier to look at); it’s fine long as you use them at least once!</p>
<ul>
<li><p>Depending on your datasets, it might be a good idea to do this before joining. For example, if you have a dataset you like with multiple measurements per year, but you want to join by year, you could average over your numeric variables to get means/year, do counts for your categoricals to get a counts/year, etc.</p></li>
<li><p>If your data sets are already tidy, demonstrate the use of <code>pivot_longer()/gather()</code> and <code>pivot_wider()/spread()</code> on all or part of your data at some point in this document (e.g., after you have generated summary statistics in part 3, make a table of them wide instead of long).</p></li>
</ul></li>
<li><p>Join your 2+ separate data sources into a single dataset based on a common ID variable! If you can’t find a good pair datasets to join, you may split one main dataset into two different datasets with a common ID variable in each, and then join them back together based on that common ID, but this is obviously less than ideal.</p>
<ul>
<li>You will document the type of join that you do (left/right/inner/full), including a discussion of how many observations were in each dataset, which observations in each dataset were dropped (if any) and why you chose this particular join.</li>
</ul></li>
<li><p>Create summary statistics</p>
<ul>
<li><p>Use <em>all six</em> core <code>dplyr</code> functions (<code>filter, select, arrange, group_by, mutate, summarize</code>) to manipulate and explore your dataset. For mutate, create a new variable that is a function of at least one other variable, preferably using a dplyr vector function (see dplyr cheatsheet). It’s totally fine to use the <code>_if</code>, <code>_at</code>, <code>_all</code> versions of mutate/summarize instead (indeed, it is encouraged if you have lots of variables)</p></li>
<li><p>Create summary statistics (<code>mean, sd, var, n, quantile, min, max, n_distinct, cor</code>, etc) for each of your numeric variables both overall and after grouping by one of your categorical variables (either together or one-at-a-time; if you have two categorical variables, try to include at least one statistic based on a grouping of two categorical variables simultaneously). If you do not have any categorical variables, create one using mutate (e.g., with <code>case_when</code> or <code>ifelse</code>) to satisfy the <code>group_by</code> requirements above. Ideally, you will find a way to show these summary statistics in an easy-to-read table (e.g., by reshaping). (You might explore the kable package for making pretty tables!) If you have lots of numeric variables (e.g., 10+), or your categorical variables have too many categories, just pick a few (either numeric variables or categories of a categorical variable) and summarize based on those. It would be a good idea to show a correlation matrix for your numeric variables (you will need it to make one of your plots).</p></li>
</ul></li>
<li><p>Make visualizations (three plots)</p>
<ul>
<li>Make a correlation heatmap of your numeric variables</li>
<li>Create at least two additional plots of your choice with ggplot that highlight some of the more interesting features of your data.</li>
<li>Each plot (besides the heatmap) should have at least three variables mapped to separate aesthetics</li>
<li>Each should use different geoms (e.g., don’t do two geom_bars)</li>
<li>At least one plot should include <code>stat="summary"</code></li>
<li>Each plot should include a supporting paragraph describing the relationships that are being visualized and any trends that are apparent
<ul>
<li>It is fine to include more, but limit yourself to 4. Plots should avoid being redundant! Four bad plots will get a lower grade than two good plots, all else being equal.</li>
</ul></li>
<li>Make them pretty! Use correct labels, etc.</li>
</ul></li>
<li><p>Perform k-means/PAM clustering or PCA on (at least) your numeric variables.</p>
<ul>
<li><p>Include all steps as we discuss in class, including a visualization.</p></li>
<li><p>If you don’t have at least 3 numeric variables, or you want to cluster based on categorical variables too, convert them to factors in R, generate Gower’s dissimilarity matrix on the data, and do PAM clustering on the dissimilarities.</p></li>
<li><p>Show how you chose the final number of clusters/principal components</p></li>
<li><p>Interpret the final clusters/principal components</p></li>
<li><p>For every step, document what your code does (in words) and what you see in the data!</p></li>
</ul></li>
</ol>
<P style="page-break-before: always">
<div style="page-break-after: always;"></div>
</div>
<div id="rubric" class="section level3">
<h3>Rubric</h3>
<p>Prerequisite: Finding appropriate data from at least two sources per the instructions above: Failure to do this will result in a 0! You will submit a .Rmd file and a knitted document (html/pdf).</p>
<div id="introduction-4-pts" class="section level4">
<h4>0. Introduction (4 pts)</h4>
<ul>
<li>Write a narrative introductory paragraph or two describing the datasets you have chosen, the variables they contain, how they were acquired, and why they are interesting to you. Expand on potential associations you may expect, if any.</li>
</ul>
<p>I got these two data sets from Cancer Cell Line Encyclopedia from the Broad Institute, which is a cancer cell line database. It includes nearly 1800 cell line samples, but my research is about breast cancer and colorectal cancer. Therefore, I only downloaded 60 breast cancer cell line data and 63 colorectal cancer. My data set 1 is cell line information including sample_ID, mutation count, age, sample type and genome altered. My second data set is gene expression levels, I only included my interested genes HK2, P63, H2AFX and KEAP1, they are genes which our group is working on. The two data sets share the same sample_ID. I chose this data set because I graduate research project is about breast cancer and H2AX, so I think this database will help me to understand about my research.</p>
<pre class="r"><code>clinical &lt;- read.table(&quot;ccle_broad_2019_clinical_data.tsv&quot;, header = T, 
    sep = &quot;\t&quot;)
mRNA &lt;- read.table(&quot;mRNA.txt&quot;, header = T, sep = &quot;\t&quot;)</code></pre>
</div>
<div id="tidying-rearranging-widelong-8-pts" class="section level4">
<h4>1. Tidying: Rearranging Wide/Long (8 pts)</h4>
<ul>
<li>Tidy the datasets (using the <code>tidyr</code> functions <code>pivot_longer</code>/<code>gather</code> and/or <code>pivot_wider</code>/<code>spread</code>)</li>
<li>If you data sets are already tidy, be sure to use those functions somewhere else in your project (e.g., for rearranging summary statistics)</li>
<li>Document the process (describe in words what was done)</li>
</ul>
<p>My data sets were already tidy, I used it in correlation plot.</p>
</div>
<div id="joiningmerging-8-pts" class="section level4">
<h4>2. Joining/Merging (8 pts)</h4>
<ul>
<li>Join your datasets into one using a <code>dplyr</code> join function</li>
<li>If you have multiple observations on the joining variable in either dataset, fix this by collapsing via summarize</li>
<li>Discuss the process in words, including why you chose the join you did</li>
<li>Discuss which cases were dropped, if any, and potential problems with this</li>
</ul>
<pre class="r"><code>library(dplyr)
combine &lt;- left_join(clinical, mRNA, by = c(Sample_ID = &quot;SAMPLE_ID&quot;))
glimpse(combine)</code></pre>
<pre><code>## Rows: 123
## Columns: 11
## $ Sample_ID      &lt;chr&gt; &quot;AU565_BREAST&quot;, &quot;BT20_BREAST&quot;, &quot;BT474_BREAST&quot;, &quot;BT483_B…
## $ Age            &lt;int&gt; 43, 74, 60, 23, 72, 72, 43, 58, 45, 35, 51, NA, NA, NA,…
## $ Genome_Altered &lt;dbl&gt; 0.7080, 0.9378, 0.6866, 0.6357, 0.5853, 0.4110, 0.5480,…
## $ Mutation_Count &lt;int&gt; 559, 416, 1025, 531, 200, 319, 181, 577, 824, 216, 455,…
## $ Sample_Type    &lt;chr&gt; &quot;Metastasis&quot;, &quot;Primary&quot;, &quot;Primary&quot;, &quot;Primary&quot;, &quot;Primary…
## $ Tumor_Type     &lt;chr&gt; &quot;breast&quot;, &quot;breast&quot;, &quot;breast&quot;, &quot;breast&quot;, &quot;breast&quot;, &quot;colo…
## $ STUDY_ID       &lt;chr&gt; &quot;ccle_broad_2019&quot;, &quot;ccle_broad_2019&quot;, &quot;ccle_broad_2019&quot;…
## $ HK2            &lt;dbl&gt; 1.4764, -0.7664, 0.5178, 0.5654, 0.7627, 0.1286, 0.4232…
## $ KEAP1          &lt;dbl&gt; -2.8521, -0.4570, 0.3060, -2.1686, 0.2310, -0.0885, -0.…
## $ H2AFX          &lt;dbl&gt; -0.3328, 0.9800, 0.1965, -0.7191, -0.0647, 0.9853, 0.14…
## $ TP63           &lt;dbl&gt; -0.4770, -0.0948, -0.1020, -0.4946, -0.3974, -0.5109, -…</code></pre>
<p>Here I used left join function from ‘dplyr’ to combine my two data sets by Sample_ID which is also named as SAMPLE_ID. I included all the information here, since I have filtered the interested genes when I downloaded it.</p>
</div>
<div id="wrangling-36-pts" class="section level4">
<h4>3. Wrangling (36 pts)</h4>
<ul>
<li>Use all six core <code>dplyr</code> functions in the service of generating summary tables/statistics (12 pts)
<ul>
<li>Use mutate at least once to generate a variable that is a function of at least one other variable</li>
</ul></li>
</ul>
<pre class="r"><code>combine %&gt;% mutate(mutation_burden = Genome_Altered * Mutation_Count) %&gt;% 
    select(Sample_ID, Tumor_Type, mutation_burden) %&gt;% arrange(-mutation_burden) %&gt;% 
    glimpse()</code></pre>
<pre><code>## Rows: 123
## Columns: 3
## $ Sample_ID       &lt;chr&gt; &quot;LS411N_LARGE_INTESTINE&quot;, &quot;CCK81_LARGE_INTESTINE&quot;, &quot;HC…
## $ Tumor_Type      &lt;chr&gt; &quot;colorectal&quot;, &quot;colorectal&quot;, &quot;breast&quot;, &quot;colorectal&quot;, &quot;c…
## $ mutation_burden &lt;dbl&gt; 1529.8150, 1217.0756, 1180.4760, 1148.6400, 1073.7909,…</code></pre>
<pre class="r"><code>combine %&gt;% filter(!Sample_Type == &quot;&quot;) %&gt;% group_by(Tumor_Type, 
    Sample_Type) %&gt;% summarise_at(c(&quot;HK2&quot;, &quot;KEAP1&quot;, &quot;H2AFX&quot;, 
    &quot;TP63&quot;), mean, na.rm = TRUE)</code></pre>
<pre><code>## # A tibble: 4 x 6
## # Groups:   Tumor_Type [2]
##   Tumor_Type Sample_Type       HK2   KEAP1   H2AFX   TP63
##   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt;
## 1 breast     Metastasis  -0.000900 -0.298   0.175  -0.316
## 2 breast     Primary     -0.0889   -0.455  -0.540  -0.188
## 3 colorectal Metastasis   0.00122   0.0991  0.104  -0.400
## 4 colorectal Primary      0.209     0.162   0.0755 -0.368</code></pre>
<ul>
<li>Compute summary statistics for each of your variables using <code>summarize</code> alone and with <code>group_by</code> (if you have more than 10 variables, fine to just focus on 10) (20 pts)
<ul>
<li>Use at least 5 unique functions inside of summarize (e.g., mean, sd)</li>
<li>For at least 2, use summarize after grouping by a categorical variable. Create one by dichotomizing a numeric if necessary</li>
<li>If applicable, at least 1 of these should group by two categorical variables</li>
</ul></li>
</ul>
<pre class="r"><code>combine %&gt;% filter(!Sample_Type == &quot;&quot;) %&gt;% group_by(Tumor_Type, 
    Sample_Type) %&gt;% summarise_at(c(&quot;HK2&quot;, &quot;KEAP1&quot;, &quot;H2AFX&quot;, 
    &quot;TP63&quot;, &quot;Age&quot;, &quot;Mutation_Count&quot;, &quot;Genome_Altered&quot;), mean, 
    na.rm = TRUE)</code></pre>
<pre><code>## # A tibble: 4 x 9
## # Groups:   Tumor_Type [2]
##   Tumor_Type Sample_Type       HK2   KEAP1   H2AFX   TP63   Age Mutation_Count
##   &lt;chr&gt;      &lt;chr&gt;           &lt;dbl&gt;   &lt;dbl&gt;   &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt;
## 1 breast     Metastasis  -0.000900 -0.298   0.175  -0.316  49.5           427.
## 2 breast     Primary     -0.0889   -0.455  -0.540  -0.188  51.7           373.
## 3 colorectal Metastasis   0.00122   0.0991  0.104  -0.400  60.2           693.
## 4 colorectal Primary      0.209     0.162   0.0755 -0.368  57.5          2155.
## # … with 1 more variable: Genome_Altered &lt;dbl&gt;</code></pre>
<pre class="r"><code>combine %&gt;% filter(!Sample_Type == &quot;&quot;) %&gt;% group_by(Tumor_Type, 
    Sample_Type) %&gt;% summarise_at(c(&quot;HK2&quot;, &quot;KEAP1&quot;, &quot;H2AFX&quot;, 
    &quot;TP63&quot;, &quot;Age&quot;, &quot;Mutation_Count&quot;, &quot;Genome_Altered&quot;), sd, na.rm = TRUE)</code></pre>
<pre><code>## # A tibble: 4 x 9
## # Groups:   Tumor_Type [2]
##   Tumor_Type Sample_Type   HK2 KEAP1 H2AFX  TP63   Age Mutation_Count
##   &lt;chr&gt;      &lt;chr&gt;       &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt;
## 1 breast     Metastasis  1.02  1.03  0.967 0.333  8.08           258.
## 2 breast     Primary     1.05  1.18  0.991 0.742 15.1            392.
## 3 colorectal Metastasis  0.921 0.610 0.874 0.415 13.8            842.
## 4 colorectal Primary     0.809 0.720 0.832 0.559 13.2           3332.
## # … with 1 more variable: Genome_Altered &lt;dbl&gt;</code></pre>
<pre class="r"><code>combine %&gt;% filter(!Sample_Type == &quot;&quot;) %&gt;% group_by(Tumor_Type) %&gt;% 
    summarise_at(c(&quot;HK2&quot;, &quot;KEAP1&quot;, &quot;H2AFX&quot;, &quot;TP63&quot;, &quot;Age&quot;, &quot;Mutation_Count&quot;, 
        &quot;Genome_Altered&quot;), median, na.rm = TRUE)</code></pre>
<pre><code>## # A tibble: 2 x 8
##   Tumor_Type   HK2  KEAP1  H2AFX   TP63   Age Mutation_Count Genome_Altered
##   &lt;chr&gt;      &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt;  &lt;dbl&gt; &lt;dbl&gt;          &lt;dbl&gt;          &lt;dbl&gt;
## 1 breast     0.136 -0.259 -0.250 -0.443    49           334.          0.631
## 2 colorectal 0.205  0.237  0.318 -0.511    56           414           0.397</code></pre>
<pre class="r"><code>combine %&gt;% filter(Tumor_Type == &quot;breast&quot;) %&gt;% summarise(H2AFX_q = quantile(H2AFX, 
    c(0.25, 0.5, 0.75), na.rm = TRUE))</code></pre>
<pre><code>##     H2AFX_q
## 1 -0.964800
## 2 -0.272150
## 3  0.736875</code></pre>
<pre class="r"><code># combine %&gt;% filter(!Sample_Type==&#39;&#39;) %&gt;%
# group_by(Tumor_Type, Sample_Type) %&gt;% summarise(n())</code></pre>
<ul>
<li>Summarize/discuss all (or the most interesting) results in no more than two paragraphs (4 pts)</li>
</ul>
<p>For my proposal work, I hypothesize H2AX loss repress distant metastasis. In this analysis it is interesting that H2AX had a higher mean expression level at metastasis site in both breast cancer and slight higher in metastatic colorectal cancer. H2AFX is a DNA damage response gene, higher expression of this gene in metastatic sites indicated metastatic cancer enduring more indigenous DNA damages.
H2AFX AND KEAP1 had a higher expression level in colorectal cancer. Colorectal cacner also had more mutations.</p>
</div>
<div id="visualizing-30-pts" class="section level4">
<h4>4. Visualizing (30 pts)</h4>
<ul>
<li>Create a correlation heatmap of your numeric variables the way we did in class</li>
</ul>
<pre class="r"><code>library(tidyr)
library(ggplot2)
cormat &lt;- combine %&gt;% filter(Tumor_Type == &quot;breast&quot;) %&gt;% na.omit() %&gt;% 
    select_if(is.numeric) %&gt;% cor(use = &quot;pair&quot;)

tidycor &lt;- cormat %&gt;% as.data.frame %&gt;% tibble::rownames_to_column(&quot;var1&quot;) %&gt;% 
    pivot_longer(-1, names_to = &quot;var2&quot;, values_to = &quot;correlation&quot;)

tidycor %&gt;% ggplot(aes(var1, var2, fill = correlation)) + geom_tile() + 
    scale_fill_gradient2(low = &quot;red&quot;, mid = &quot;white&quot;, high = &quot;blue&quot;) + 
    geom_text(aes(label = round(correlation, 2)), color = &quot;black&quot;, 
        size = 4) + theme(axis.text.x = element_text(angle = 90, 
    hjust = 1)) + coord_fixed()</code></pre>
<p><img src="/project/project1_files/figure-html/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" />
- Create two effective, polished plots with ggplot</p>
<pre><code>- Each plot should map 3+ variables to aesthetics 
- Each plot should have a title and clean labeling for all mappings
- Change at least one default theme element and color for at least one mapping per plot
- For at least one plot, add more tick marks (x, y, or both) than are given by default
- For at least one plot, use the stat=&quot;summary&quot; function
- Supporting paragraph or two (for each plot) describing the relationships/trends that are apparent</code></pre>
<pre class="r"><code>H2AFX_plot &lt;- combine %&gt;% filter(!Sample_Type == &quot;&quot;) %&gt;% filter(!is.na(H2AFX))
stat_box_data &lt;- function(y, upper_limit = max(H2AFX_plot$H2AFX) * 
    1.15) {
    return(data.frame(y = 0.95 * upper_limit, label = paste(&quot;count =&quot;, 
        length(y), &quot;\n&quot;, &quot;mean =&quot;, round(mean(y), 1), &quot;\n&quot;)))
}
ggplot(H2AFX_plot, aes(x = Sample_Type, y = H2AFX, fill = Sample_Type)) + 
    geom_boxplot(width = 0.3) + stat_summary(fun.data = stat_box_data, 
    geom = &quot;text&quot;, hjust = 0.5, vjust = 0.9, size = 3) + facet_wrap(vars(Tumor_Type)) + 
    xlab(&quot;&quot;) + theme_classic() + theme(axis.text = element_text(size = 15, 
    face = &quot;bold&quot;), axis.title = element_text(size = 20, face = &quot;bold.italic&quot;))</code></pre>
<p><img src="/project/project1_files/figure-html/unnamed-chunk-6-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>KEAP1_plot &lt;- combine %&gt;% filter(!Sample_Type == &quot;&quot;) %&gt;% filter(!is.na(KEAP1))
stat_box_data &lt;- function(y, upper_limit = max(KEAP1_plot$KEAP1) * 
    1.15) {
    return(data.frame(y = 0.95 * upper_limit, label = paste(&quot;count =&quot;, 
        length(y), &quot;\n&quot;, &quot;mean =&quot;, round(mean(y), 1), &quot;\n&quot;)))
}
ggplot(KEAP1_plot, aes(x = Sample_Type, y = KEAP1, fill = Sample_Type)) + 
    geom_boxplot(width = 0.3) + stat_summary(fun.data = stat_box_data, 
    geom = &quot;text&quot;, hjust = 0.5, vjust = 0.9, size = 3) + facet_wrap(vars(Tumor_Type)) + 
    xlab(&quot;&quot;) + theme(axis.text = element_text(size = 15, face = &quot;bold&quot;), 
    axis.title = element_text(size = 20, face = &quot;bold.italic&quot;))</code></pre>
<p><img src="/project/project1_files/figure-html/unnamed-chunk-6-2.png" width="672" style="display: block; margin: auto;" />
In my first plot, I analyzed H2AFX expression level in the two cancer types, it showed that only in breast cancer which has significant difference between metastasis and primary tumor. However, unlike H2AFX, another metabolic gene KEAP1 in my second plot didn’t show a expression differences between metastasis and primary tumor.</p>
</div>
<div id="dimensionality-reduction-20-pts" class="section level4">
<h4>5. Dimensionality Reduction (20 pts)</h4>
<ul>
<li><p>Either k-means/PAM clustering or PCA (inclusive “or”) should be performed on at least three of your variables (3 is just the minimum: using more/all of them will make this much more interesting!)</p>
<ul>
<li>All relevant steps discussed in class</li>
<li>A visualization of the clusters or the first few principal components (using ggplot2)</li>
<li>Supporting paragraph or two describing results found, interpreting the clusters/PCs etc.</li>
</ul></li>
</ul>
<pre class="r"><code>### k-means/PAM library(cluster)
K_H2AFX &lt;- H2AFX_plot %&gt;% select_if(is.numeric) %&gt;% na.omit()
### for(i in 2:10){ kms &lt;- kmeans(K_H2AFX,centers=i) sil &lt;-
### silhouette(kms$cluster,dist(K_H2AFX))
### sil_width[i]&lt;-mean(sil[,3]) }
### ggplot()+geom_line(aes(x=1:10,y=sil_width))+scale_x_continuous(name=&#39;k&#39;,breaks=1:10)
### pam1 &lt;- K_H2AFX %&gt;% pam(k=2) pamclust&lt;-K_H2AFX %&gt;%
### mutate(cluster=as.factor(pam1$clustering)) pamclust %&gt;%
### ggplot(aes(H2AFX,HK2,color=cluster)) + geom_point()

### PCA
H2AFX_num &lt;- K_H2AFX %&gt;% scale
tmp &lt;- H2AFX_plot %&gt;% na.omit()
rownames(H2AFX_num) &lt;- tmp$Sample_ID
H2AFX_pca &lt;- princomp(H2AFX_num)
summary(H2AFX_pca, loadings = T)</code></pre>
<pre><code>## Importance of components:
##                          Comp.1    Comp.2    Comp.3    Comp.4    Comp.5
## Standard deviation     1.311704 1.1331862 1.0421762 0.9848107 0.8686395
## Proportion of Variance 0.248557 0.1855056 0.1569050 0.1401070 0.1090018
## Cumulative Proportion  0.248557 0.4340626 0.5909676 0.7310746 0.8400764
##                            Comp.6     Comp.7
## Standard deviation     0.81666348 0.66339092
## Proportion of Variance 0.09634756 0.06357604
## Cumulative Proportion  0.93642396 1.00000000
## 
## Loadings:
##                Comp.1 Comp.2 Comp.3 Comp.4 Comp.5 Comp.6 Comp.7
## Age             0.114  0.297  0.467  0.730  0.327         0.194
## Genome_Altered  0.595 -0.104 -0.255 -0.193  0.280         0.673
## Mutation_Count -0.630         0.108        -0.284         0.707
## HK2            -0.224 -0.562  0.309 -0.186  0.530  0.467       
## KEAP1          -0.405  0.179 -0.470         0.651 -0.397       
## H2AFX          -0.144  0.667 -0.135 -0.247         0.671       
## TP63                  -0.316 -0.609  0.577 -0.161  0.410</code></pre>
<pre class="r"><code>eigval &lt;- H2AFX_pca$sdev^2
varprop = round(eigval/sum(eigval), 2)
ggplot() + geom_bar(aes(y = varprop, x = 1:7), stat = &quot;identity&quot;) + 
    xlab(&quot;&quot;) + geom_path(aes(y = varprop, x = 1:7)) + geom_text(aes(x = 1:7, 
    y = varprop, label = round(varprop, 2)), vjust = 1, col = &quot;white&quot;, 
    size = 5) + scale_y_continuous(breaks = seq(0, 0.6, 0.2), 
    labels = scales::percent) + scale_x_continuous(breaks = 1:7)</code></pre>
<p><img src="/project/project1_files/figure-html/unnamed-chunk-7-1.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>H2AFXdf &lt;- data.frame(Name = tmp$Sample_ID, PC1 = H2AFX_pca$scores[, 
    1], PC2 = H2AFX_pca$scores[, 2], group = tmp$Tumor_Type)
ggplot(H2AFXdf, aes(PC1, PC2, color = group)) + geom_point()</code></pre>
<p><img src="/project/project1_files/figure-html/unnamed-chunk-7-2.png" width="672" style="display: block; margin: auto;" /></p>
<pre class="r"><code>H2AFXdf_2 &lt;- data.frame(Name = tmp$Sample_ID, PC2 = H2AFX_pca$scores[, 
    2], PC3 = H2AFX_pca$scores[, 3], group = tmp$Tumor_Type)
ggplot(H2AFXdf_2, aes(PC2, PC3, color = group)) + geom_point()</code></pre>
<p><img src="/project/project1_files/figure-html/unnamed-chunk-7-3.png" width="672" style="display: block; margin: auto;" />
I tried both k-mean and PCA to do the classification. PCA showed a better classification for tumor type. I used the first two components to do the analysis, since they account for 44% of my data. But this is still no clear difference with all the numeric variables in my data. I think more genes are needed for this classification.</p>
</div>
<div id="neatness-holisticdiscretionary-points-4-pts" class="section level4">
<h4>6. Neatness, Holistic/Discretionary Points (4 pts)</h4>
<ul>
<li>Keep things looking nice! Your project should not knit to more than 30 or so pages (probably closer to 10-20)! You will lose points if you print out your entire dataset(s), have terrible formatting, etc. If you start your project in a fresh .Rmd file, you are advised to copy set-up code below and include it: this will do things like automatically truncate if you accidentally print out a huge dataset, etc. Imagine this is a polished report you are giving to your PI or boss to summarize your work researching a topic.</li>
</ul>
<pre class="r"><code>## paste this chunk into the ```{r setup} chunk at the top of
## your project 1 .Rmd file

knitr::opts_chunk$set(echo = TRUE, eval = TRUE, fig.align = &quot;center&quot;, 
    warning = F, message = F, tidy = TRUE, tidy.opts = list(width.cutoff = 60), 
    R.options = list(max.print = 100))</code></pre>
</div>
</div>
<div id="where-do-i-find-data" class="section level3">
<h3>Where do I find data?</h3>
<p>OK, brace yourself!</p>
<p>You can choose ANY datasets you want that meet the above criteria for variables and observations. I’m just sitting here but off the top of my head, if you are into amusement parks, you could look at amusement-park variables, including ticket sales per day etc.; then you could join this by date in weather data. If you are interested in Game of Thrones, you could look at how the frequency of mentions of character names (plus other character variables) and the frequency of baby names in the USA…You could even take your old Biostats data and merge in new data (e.g., based on a Google forms timestamp).</p>
<p>You could engage in some “me-search”: You can <a href="https://support.spotify.com/ca-en/article/data-rights-and-privacy-settings/">request your Spotify data</a> or <a href="https://help.netflix.com/en/node/101917">download Netflix viewing activity</a>, Amazon purchase history, etc. You can use your Google Fit/Fitbit/Apple watch data, etc. These can be combined (e.g., with each other, with other data sources).</p>
<p>You can make it as serious as you want, or not, but keep in mind that you will be incorporating this project into a portfolio webpage for your final in this course, so choose something that really reflects who you are, or something that you feel will advance you in the direction you hope to move career-wise, or something that you think is really neat. On the flip side, regardless of what you pick, you will be performing all the same tasks, so it doesn’t end up being that big of a deal.</p>
<p>If you are totally clueless and have no direction at all, log into the server and type</p>
<pre class="r"><code>data(package = .packages(all.available = TRUE))</code></pre>
<p>This will print out a list of <strong>ALL datasets in ALL packages</strong> installed on the server (a ton)! Scroll until your eyes bleed! Actually, do not scroll that much… To start with something more manageable, just run the command on your own computer, or just run <code>data()</code> to bring up the datasets in your current environment. To read more about a dataset, do <code>?packagename::datasetname</code>.</p>
<p>If it is easier for you, and in case you don’t have many packages installed, a list of R datasets from a few common packages (also downloadable in CSV format) is given at the following website: <a href="https://vincentarelbundock.github.io/Rdatasets/datasets.html" class="uri">https://vincentarelbundock.github.io/Rdatasets/datasets.html</a> (including types/numbers of variables in each)</p>
<ul>
<li><p>A good package to download for fun/relevant data is <code>fivethiryeight</code>. Just run <code>install.packages("fivethirtyeight"), load the packages with</code>library(fivethirtyeight)<code>, run</code>data()`, and then scroll down to view the datasets. Here is an online list of all 127 datasets (with links to the 538 articles). Lots of sports, politics, current events, etc: <a href="https://cran.r-project.org/web/packages/fivethirtyeight/vignettes/fivethirtyeight.html" class="uri">https://cran.r-project.org/web/packages/fivethirtyeight/vignettes/fivethirtyeight.html</a></p></li>
<li><p>If you have already started to specialize (e.g., ecology, epidemiology) you might look at discipline-specific R packages (vegan, epi, respectively). We will be using some tools from these packages later in the course, but they come with lots of data too, which you can explore according to the directions above</p></li>
<li><p>However, you <em>emphatically DO NOT</em> have to use datasets available via R packages! In fact, I would much prefer it if you found the data from completely separate sources and brought them together (a much more realistic experience in the real world)! You can even reuse data from your SDS328M project, provided it shares a variable in common with other data which allows you to merge the two together (e.g., if you still had the timestamp, you could look up the weather that day: <a href="https://www.wunderground.com/history/" class="uri">https://www.wunderground.com/history/</a>). If you work in a research lab or have access to old data, you could potentially merge it with new data from your lab!</p></li>
<li><p>Here is a curated list of interesting datasets (read-only spreadsheet format): <a href="https://docs.google.com/spreadsheets/d/1wZhPLMCHKJvwOkP4juclhjFgqIY8fQFMemwKL2c64vk/edit" class="uri">https://docs.google.com/spreadsheets/d/1wZhPLMCHKJvwOkP4juclhjFgqIY8fQFMemwKL2c64vk/edit</a></p></li>
<li><p>Here is another great compilation of datasets: <a href="https://github.com/rfordatascience/tidytuesday" class="uri">https://github.com/rfordatascience/tidytuesday</a></p></li>
<li><p>Here is the UCI Machine Learning Repository: <a href="https://archive.ics.uci.edu/ml/index.php" class="uri">https://archive.ics.uci.edu/ml/index.php</a></p>
<ul>
<li>See also <a href="https://en.wikipedia.org/wiki/List_of_datasets_for_machine-learning_research#Biological_data" class="uri">https://en.wikipedia.org/wiki/List_of_datasets_for_machine-learning_research#Biological_data</a></li>
</ul></li>
<li><p>Here is another good general place to look: <a href="https://www.kaggle.com/datasets" class="uri">https://www.kaggle.com/datasets</a></p></li>
<li><p>To help narrow your search down or to see interesting variable ideas, check out <a href="https://www.tylervigen.com/spurious-correlations" class="uri">https://www.tylervigen.com/spurious-correlations</a>. This is the spurious correlations website, and it is fun, but if you look at the bottom of each plot you will see sources for the data. This is a good place to find very general data (or at least get a sense of where you can scrape data together from)!</p></li>
<li><p>If you are interested in medical data, check out www.countyhealthrankings.org</p></li>
<li><p>If you are interested in scraping UT data, the university makes <em>loads</em> of data public (e.g., beyond just professor CVs and syllabi). Check out all the data that is available in the statistical handbooks: <a href="https://reports.utexas.edu/statistical-handbook" class="uri">https://reports.utexas.edu/statistical-handbook</a></p></li>
</ul>
<div id="broader-data-sources" class="section level5">
<h5>Broader data sources:</h5>
<p><a href="www.data.gov">Data.gov</a> 186,000+ datasets!</p>
<p><a href="Social%20Explorer">Social Explorer</a> is a nice interface to Census and American Community Survey data (more user-friendly than the government sites). May need to sign up for a free trial.</p>
<p><a href="www.bls.gov">U.S. Bureau of Labor Statistics</a></p>
<p><a href="www.census.gov">U.S. Census Bureau</a></p>
<p><a href="www.gapminder.org/data">Gapminder</a>, data about the world.</p>
<p>…</p>
</div>
</div>
</div>

            
        <hr>         <div class="related-posts">
                <h5>Related Posts</h5>
                
              </div> 
            </div>
          </div>

   <hr>  <div class="disqus">
  <div id="disqus_thread"></div>
  <script type="text/javascript">

    (function() {
      
      
      if (window.location.hostname == "localhost")
        return;

      var disqus_shortname = '';
      var dsq = document.createElement('script'); dsq.type = 'text/javascript'; dsq.async = true;
      dsq.src = '//' + disqus_shortname + '.disqus.com/embed.js';
      (document.getElementsByTagName('head')[0] || document.getElementsByTagName('body')[0]).appendChild(dsq);
    })();
  </script>
  <noscript>Please enable JavaScript to view the <a href="http://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
  <a href="http://disqus.com/" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>
</div> 
        </div>
      </div>
    </div>

    
    <footer>
  <div id="footer">
    <div class="container">
      <p class="text-muted">&copy; All rights reserved. Powered by <a href="https://gohugo.io/">Hugo</a> and
      <a href="http://www.github.com/nurlansu/hugo-sustain/">sustain</a> with ♥</p>
    </div>
  </div>
</footer>
<div class="footer"></div>


<script src="https://ajax.googleapis.com/ajax/libs/jquery/1.11.3/jquery.min.js"></script>

<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script>
<script src="/js/docs.min.js"></script>
<script src="/js/main.js"></script>

<script src="/js/ie10-viewport-bug-workaround.js"></script>


    
  </body>
</html>
